fdbclient/NativeAPI.actor.cpp

```diff
@@ -224,10 +225,16 @@ void DatabaseContext::getLatestCommitVersions(const Reference<LocationInfo>& loc
 	}
 
 	if (ssVersionVectorCache.getMaxVersion() != invalidVersion && readVersion > ssVersionVectorCache.getMaxVersion()) {
-		TraceEvent(SevError, "GetLatestCommitVersions")
+		// This can happen after a recovery, GRV proxy change is detected and
+		// version cache is cleared. However, an old GRV response is processed
+		// to update the version cache with a relatively old version. Then
+		// another transaction with a newer read version from the old GRV proxy
+		// issues a read, which can trigger the condition here. Throw here to
+		// restart the transaction that will get a new GRV from the new proxy.
+		TraceEvent(SevWarn, "GetLatestCommitVersions")
 		    .detail("ReadVersion", readVersion)
-		    .detail("Version vector", ssVersionVectorCache.toString());
-		ASSERT(false);
+		    .detail("VersionVector", ssVersionVectorCache.toString());
+		throw transaction_too_old();
```

Comment:
Normally I thought "transaction too old" meant a read was done in a translation that lasted longer than 5 seconds. Is this the right error to throw?

Reply:
I agree `transaction_too_old` is not accurate. Since the effect is the same, so I was lazy.

Ideally we could add a new error code like `stale_version_vector` and do proper `onError` handling like the `transaction_too_old` error. I am fine adding a new error code. What do you think?

Reply:
If we plan to merge vv to main branch soon, I think we should - it seems the right answer if not too difficult.

Reply:
Done.